app/src/main/java/com/example/hm3_retrofit/fragments/ListFragment.kt

```diff
@@ -0,0 +1,124 @@
+package com.example.hm3_retrofit.fragments
+
+import retrofit2.*
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.hm3_retrofit.R
+import com.example.hm3_retrofit.adapter.ItemAdapter
+import com.example.hm3_retrofit.databinding.FragmentListBinding
+import com.example.hm3_retrofit.model.ItemType
+import com.example.hm3_retrofit.model.ResponseApi
+import com.example.hm3_retrofit.retrofit.RickMortyService
+
+
+class ListFragment : Fragment() {
+
+    private var _binding: FragmentListBinding? = null
+    private val binding: FragmentListBinding
+        get() = requireNotNull(_binding) {
+            "View was destroyed"
+        }
+
+    private val myAdapter by lazy {
+        ItemAdapter { item ->
+            val personItem = item as? ItemType.CartoonPerson ?: return@ItemAdapter
+            findNavController().navigate(
+                ListFragmentDirections.toDetails(personItem.idApi)
+            )
+        }
+    }
+
+    private var requesrCall: Call<ResponseApi>? = null
+    private var pageCounter = 1
+    private var isLoading = false
+    private var finalFResultlist: List<ItemType> = emptyList()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        return FragmentListBinding.inflate(inflater, container, false)
+            .also { _binding = it }
+            .root
+
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val layoutManager = LinearLayoutManager(view.context)
+        //recycler init
+        with(binding) {
+            recyclerView.addSpaceDecoration(resources.getDimensionPixelSize(R.dimen.bottom_space))
+            recyclerView.adapter = myAdapter
+            recyclerView.layoutManager = layoutManager
+        }
+
+        swipeRefresh()
+        makeRequest(1)
+
+        with(binding) {
+            recyclerView.addPaginationScrollListener(layoutManager, 2) {
+                if (!isLoading) {
+                    isLoading = true
+                    makeRequest(pageCounter++)
+                    isLoading = false
+                }
+            }
+        }
+    }
+
+    private fun swipeRefresh() {
+        binding.swipeLayout.setOnRefreshListener {
+            //todo допилить обнуление общего листа по свайпу
+            pageCounter=1
+            makeRequest(pageCounter)
```

Comment:
сюда должна передаваться лямбда для того чтобы в колбеке когда пришел результат от сервера выполнять уже действие по скрытию swipeRefresh
```suggestion
            makeRequest(pageCounter) { binding.swipeLayout.isRefreshing = false }
```